### DISCUSSION TITLE
Close issues with a reason / status

### DISCUSSION THREAD

**Discussion by Zerotask** [community/community#4514] [Projects and Issues] [answered]
At the moment you can't see if an issue was resolved, fixed etc., you just see that it is closed.
This means if you encounter a similar problem you have to open the issue to get the conclusion / reason why it was closed.

A work-around for this case is using labels, although it would be better to have an option to set a close reason or status, e. g. Fixed / Answered (the latter is similar to discussions - you see a green check in the overview list).

Similar to discussions, the issue list can then show small icons for that. This way you can see the conclusion of closed issues without opening them.

**Comment by thedrow** [comment]
I'd be satisfied with labels.

**Comment by stingalleman** [comment]
This would be sick!

**Comment by Zerotask** [comment]
An update to this is currently in the preview state 🥳

https://github.blog/changelog/2022-03-10-the-new-github-issues-march-10th-update/

**Reply by azenMatt** [reply]
@Zerotask - let me know if you have an organisation you would like to Include in this preview :) (I can also just add your handle if you rather - but then no one else would be able to see it)

**Reply by Zerotask** [reply]
Thanks for the offer, but currently I don't have an organization, but I'm happy to test it for my personal repos

**Reply by thedrow** [reply]
@azenMatt Celery would like to try using this feature :)

There's an important missing reason which Github doesn't natively support. Sometimes a contribution becomes stale and gets superseded by another PR. I think we should have a status for that as well.

**Reply by azenMatt** [reply]
🎉 Thanks @Zerotask and @thedrow - @Zerotask I added your personal account and @thedrow I added the Celery org 😄

**Reply by francismartens318** [reply]
@azenMatt 
Can you also add Exalate-team (organisation) to the preview - thanks !

**Reply by azenMatt** [reply]
hi @francismartens318 - we just launched this to everyone on GitHub.com yesterday! :D

**Comment by azenMatt** [comment, marked as answer]
This is now live! Thanks everyone for your feedback during the preview 😄 

https://github.blog/changelog/2022-05-19-the-new-github-issues-may-19th-update/

**Reply by francisidalko** [reply]
Hehe - thanks for that!

**Reply by RyanCavanaugh** [reply]
Forcing maintainers to select a reason is introducing some friction, since we have automation and muscle memory that closes things *generally* for "not planned" reasons, e.g. https://github.com/microsoft/TypeScript/issues/49190#event-6663169263, but it gets marked as "completed" and people get annoyed at us.

It would have been nicer to have a default "no reason" and let us opt in to competed vs not planned.

**Reply by hakman** [reply]
I also have to agree that using `completed` as the default reason is not quite ideal.
Unless there is a PR fixing the issue (which should be closed with `completed`, most likely the `not planned` status would be a more likely candidate.
Already seeing some confused comments related to this too: https://github.com/kubernetes/kubernetes/issues/23130#issuecomment-1159743385.

**Reply by azenMatt** [reply]
@hakman - this looks like it will work quite well once the automated bot is updated to close as not planned?

**Reply by hakman** [reply]
@azenMatt Yes, once the bot is updated it will have the expected status. Though, having a default as `completed` adds a lot of unnecessary chat and frustration. Issues are most likely `completed` when there is a PR closing them. Otherwise, this feature should not assume `completed` by default.

**Reply by zachleat** [reply]
One more upvote that the defaults here are incompatible with my project. I use the workflow documented here: https://twitter.com/samselikoff/status/991395669016436736 Feature requests are closed but not completed. They are also _not_ not planned.

**Reply by kblok** [reply]
@azenMatt any plans to support that in the API? I was wondering whether [probot/stale](https://github.com/probot/stale) could close an issue with the not planned reason.

**Comment by odahcam** [comment]
Are we going to have more or even custom close reasons?
